includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -328,8 +330,13 @@ function ( $id ) use ( $is_test_mode ) {
 	 */
 	public static function is_enabled() {
 		$stripe_settings = WC_Stripe_Helper::get_stripe_settings();
-		$key             = WC_Stripe_Mode::is_test() ? 'test_connection_type' : 'connection_type';
-		return isset( $stripe_settings[ $key ] ) && 'connect' === $stripe_settings[ $key ];
+
+		$connection_type_key              = WC_Stripe_Mode::is_test() ? 'test_connection_type' : 'connection_type';
+		$is_account_connected_to_platform = isset( $stripe_settings[ $connection_type_key ] ) && 'connect' === $stripe_settings[ $connection_type_key ];
+
+		$is_pmc_disabled = isset( $stripe_settings['pmc_enabled'] ) && 'no' === $stripe_settings['pmc_enabled'];
+
+		return $is_account_connected_to_platform && ! $is_pmc_disabled;
```

Comment:
Non-blocking nit: I think this could be easier to read if we used early returns.
```suggestion
		// If the account is not a Connect account, we should not sync payment methods.
		if ( ! isset( $stripe_settings[ $connection_type_key ] ) || 'connect' !== $stripe_settings[ $connection_type_key ] ) {
			return false;
		}

		// If we don't have the pmc_enabled flag, or it is set to no, we should not sync payment methods.
		if ( ! isset( $stripe_settings['pmc_enabled'] ) || 'no' === $stripe_settings['pmc_enabled'] ) {
			return false;
		}

		return true;
```

Reply:
Fixed in [d353bd0](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4334/commits/d353bd0d6df8762c1528ef9c2607aecb36a6ba6a).
The pmc_enabled check needs to only disable the use of PMCs if the `pmc_enabled` flag is set and it's `no`. When not set (or empty), it means the migration was not attempted yet (that will result in a 'yes' or 'no'). I added a comment explaining this use case.

```
// If we have the pmc_enabled flag and it's set to no, we should not use payment methods configurations.
if ( isset( $stripe_settings['pmc_enabled'] ) && 'no' === $stripe_settings['pmc_enabled'] ) {
	return false;
}
```